test-utils/testlib/src/standard_test_cases.rs

```diff
@@ -240,6 +240,22 @@ pub fn test_send_money(node: impl Node) {
     );
 }
 
+pub fn test_smart_contract_reward(node: impl Node) {
+    let node_user = node.user();
+    let root = node_user.get_state_root();
+    let bob = node_user.view_account(&bob_account()).unwrap();
+    assert_eq!(bob.amount, TESTING_INIT_BALANCE - TESTING_INIT_STAKE);
+    let transaction_result =
+        node_user.function_call(alice_account(), bob_account(), "run_test", vec![], 1000000, 0);
+    assert_eq!(transaction_result.status, FinalTransactionStatus::Completed);
+    assert_eq!(transaction_result.transactions.len(), 3);
+    let new_root = node_user.get_state_root();
+    assert_ne!(root, new_root);
+
+    let bob = node_user.view_account(&bob_account()).unwrap();
+    assert!(bob.amount > TESTING_INIT_BALANCE - TESTING_INIT_STAKE);
```

Comment:
let's actually check for real balances.
If you need to turn off the fees for the other stuff - there is now RuntimeConfig::free() for 0 fees.

Reply:
i need to know exact gas balance of the function call execution before I can compute the balance.

Reply:
Is that not doable?

Reply:
The gas compute might change which would require updating the test.
It'll be easier once we return gas_burnt in TxLog